raphtory/src/core/entities/edges/edge_store.rs

```diff
@@ -95,32 +117,27 @@ impl EdgeStore {
     }
 
     pub fn internal_num_layers(&self) -> usize {
-        self.layers
-            .len()
-            .max(self.additions.len())
-            .max(self.deletions.len())
-    }
-    fn get_or_allocate_layer(&mut self, layer_id: usize) -> &mut EdgeLayer {
-        if self.layers.len() <= layer_id {
-            self.layers.resize_with(layer_id + 1, Default::default);
+        self.data.len()
+    }
+    fn get_or_allocate_layer(&mut self, layer_id: usize) -> EdgeLayerMut {
+        if self.data.len() <= layer_id {
+            self.data.resize_with(layer_id + 1, Default::default);
         }
-        &mut self.layers[layer_id]
+        EdgeLayerMut(&mut self.data[layer_id])
```

Comment:
could just return `&mut self.data[layer_id].layer` here?

Reply:
got carried away, it's gone now